### PR TITLE
Feat/cli: added mnemonic gen from console only

### LINKED ISSUE #652: cli : generate mnemonic without interactive mode
## Description

Generation of mnemonic without interaction mode
example: 
```
/wallet mnemonic –output-file-name file-name.txt or ./wallet mnemonic –output-stdout.
```

## Motivation

CLI is many times used for simple shell automation scripts. 

## Requirements

Write a list of what you want this feature to do.

1.  ./wallet mnemonic --output-file-name  filename.txt  will generate a mnemonic to the  file
2. ./wallet mnemonic --output-stdout will generate a mnemonic to the stdout
3. 

## Open questions (optional)

name of flags

## Are you planning to do it yourself in a pull request?

Yes/No.

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

Allows getting mnemonic through just command line args

## Links to any relevant issues

fixes [#652](https://github.com/iotaledger/iota-sdk/issues/652)

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Test cases:
- `./wallet mnemonic --output-stdout no` -> Write to default file **DO WE WANT THIS OR BACK TO USER CHOICE**
- `./wallet mnemonic --output-stdout yes` -> Write to console
- `./wallet mnemonic --output-stdout` -> Write to console
- `./wallet mnemonic --output-file-name filename.txt` -> Writes to file
- `./wallet mnemonic --output-file-name filename.txt --output-stdout` -> Writes to file and console
- `./wallet mnemonic` -> Prompts user

### CODE REVIEW

**Comment by thibault-martinez** [comment]
Not 2.0?

**Comment by kwek20** [comment]
oh :D
didnt even realize

File: cli/src/command/wallet.rs [hunk 1]
```diff
@@ -63,7 +63,14 @@ pub enum WalletCommand {
         path: Option<String>,
     },
     /// Generate a random mnemonic.
-    Mnemonic,
+    Mnemonic {
+        // Output the mnemonic to the specified file.
+        #[arg(long)]
+        output_file_name: Option<String>,
+        // Output the mnemonic to the stdout.
+        #[arg(long, num_args = 0..2, default_missing_value = Some("true"), value_parser = BoolishValueParser::new())]
```
Comment (DaughterOfMars) [inline cli/src/command/wallet.rs]: Does this still allow using the flag without args, like so?
``` cli --output-stdout ```

File: cli/src/helper.rs [hunk 2]
```diff
@@ -121,36 +121,54 @@ pub async fn enter_or_generate_mnemonic() -> Result<Mnemonic, Error> {
...
-        .with_prompt("Select how to proceed with it")
-        .items(&choices)
-        .default(0)
-        .interact_on(&Term::stderr())?;
+    let file_path = output_file_name
+        .clone()
+        .unwrap_or(DEFAULT_MNEMONIC_FILE_PATH.to_string());
+
+    let selected_choice = match (output_file_name, output_stdout) {
+        (None, None) => {
```
Comment (DaughterOfMars) [inline cli/src/helper.rs]: shouldn't this match `(None, Some(false))` too?

File: cli/src/command/wallet.rs (hunk 1 above)
Comment (kwek20) [reply cli/src/command/wallet.rs]: yes, its one of the test cases i did, the 0 num args allows this if there is a default

File: cli/src/helper.rs (hunk 2 above)
Comment (kwek20) [reply cli/src/helper.rs]: I chose not to, as when you provide no to console, then the user picking section still allows picking console, kinda weird.
But i left the question open in test case 1

File: cli/src/command/wallet.rs (hunk 1 above)
Comment (DaughterOfMars) [reply cli/src/command/wallet.rs]: What about the other instance above then?

Comment (DaughterOfMars) [reply cli/src/command/wallet.rs]: Also can you use `0..=1` instead? This range is a little less clear to read

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Changelog

File: cli/src/helper.rs (hunk 2 above)
Comment (DaughterOfMars) [reply cli/src/helper.rs]: Yeah, it is weird but it also doesn't really make sense to write to file when the user doesn't specify they want to

Comment (kwek20) [reply cli/src/helper.rs]: lets see what the others think

File: cli/src/command/wallet.rs (hunk 1 above)
Comment (kwek20) [reply cli/src/command/wallet.rs]: ah yea that works better

**Review by Thoralf-M — DISMISSED** [review]
I'm fine with this, will also approve again if you change the match to `(None, Some(false))`

**Review by Thoralf-M — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**